packages/serialize-request/lib/index.js

```diff
@@ -128,9 +128,10 @@ function serializeRequest(request, options = {}) {
 
 	// Serialize the headers
 	if (
-		request.headers &&
-		typeof request.headers === 'object' &&
-		!Array.isArray(request.headers) &&
+		((request.headers &&
+			typeof request.headers === 'object' &&
+			!Array.isArray(request.headers)) ||
+			isIterableHeadersObject(request.headers)) &&
 		request.headers !== null
```

Comment:
**suggestion:** this works as expected but it took me a few tries to read it, could we maybe make it a little more readable by using a variable? Something like:

```js
// Serialize the headers
const headersIsObject =
	typeof request.headers === 'object' &&
	!Array.isArray(request.headers) &&
	request.headers !== null;
if (
	request.headers &&
	(headersIsObject || isIterableHeadersObject(request.headers))
) {
	requestProperties.headers = serializeHeaders(
		request.headers,
		includeHeaders
	);
}
```